openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,839 @@
-    });
-
-    // Set custom map div height
-    $('#map').height("300px");
-
-    $('#load-project').button().click(function() {
-        $('#loadProjectDialog').dialog('open');
-    });
-
-    // Remove layer 
-    var removeLayer = function () {
-        var e = document.getElementById('layer-list');
-        var layerId = e.options[e.selectedIndex].value;
-
-        // Look up the layer id using the layer name
-        var layerIdArray = layerNames[layerId];
-        var selectedLayer = layerIdArray.toString();
-
-        // Check in the layer is in the map port
-        if (selectedLayer in layers) {
-            layerControl.removeLayer(layers[selectedLayer]);
-            map.removeLayer(layers[selectedLayer]);
-            delete layers[selectedLayer];
-        }
-        else {
-            showRemoveMsg();
-        }
-    };
-
-    // Get layer names from tilestream
-    var tileStreamLayer = '';
-    var category = '';
-    var selCat = document.getElementById('layer-category');
-    var selLayer = document.getElementById('layer-list');
-
-    // Create a header for the menu drop down
-    var catMenuHeader = document.createElement('option');
-    catMenuHeader.innerHTML = 'Projects:';
-    selCat.appendChild(catMenuHeader);
-
-    $.getJSON(TILESTREAM_API_URL,
-    function(json) {
-
-        var name, cat, type, grids, pDef;
-
-        // Create the category list (build the object)
-        for (var i=0; i < json.length; i++) {
-            name = json[i].mapped_value;
-            cat = json[i].category;
-            type = json[i].type;
-            grids = json[i].grids;
-            pDef = json[i].project_definition;
-
-            if (cat != undefined && type == 'svir-qgis') {
-                categoryList.push(cat);
-                layerNames[name] = [];
-                layersByCat[cat] = [];
-                projectDefinition[name] = [];
-
-                if (grids != undefined) {
-                    var grid = grids.toString();
-                    var gridName = grid.split('/')[4];
-                    layerGrids.push(gridName);
+ // Match each primary indicator with it's respective data value
+    var primaryRiskIndicatorObj = {};
+    // Iterate of the layer attribute features and capture the properties object
+    for (var ic = 0; ic < la.length; ic++) {
+        var tempPropertiesObj = la[ic].properties;
+        // for each layer attribute, iterate over each indicator child key
+        for (var d = 0; d < indicatorChildrenKey.length; d++) {
+            // interate over each of the properties objects
+            for (var o in tempPropertiesObj) {
+                // check for a match between the indicator child key and the properties key
+                if (indicatorChildrenKey[d] == o) {
+                    // if a match is made, capture some information about the feature
+                    // and populate the primary risk indicator object
+                    var tempValue = indicatorChildrenKey[d];
```

Comment:
The new comments look good. Now we need to rename tempValueN into something that describes their meaning.